sentry/src/main/java/io/sentry/transport/RateLimiter.java

```diff
@@ -301,4 +328,39 @@ private long parseRetryAfterOrDefault(final @Nullable String retryAfterHeader) {
     }
     return retryAfterMillis;
   }
+
+  private void notifyRateLimitObservers(final boolean applied) {
+    for (IRateLimitObserver observer : rateLimitObservers) {
+      observer.onRateLimitChanged(applied);
+    }
+  }
+
+  public void addRateLimitObserver(@NotNull final IRateLimitObserver observer) {
+    rateLimitObservers.add(observer);
+  }
+
+  public void removeRateLimitObserver(@NotNull final IRateLimitObserver observer) {
+    rateLimitObservers.remove(observer);
+  }
+
+  @Override
+  public void close() throws IOException {
+    synchronized (timerLock) {
+      if (timer != null) {
+        timer.cancel();
+        timer = null;
+      }
+    }
+  }
+
+  public interface IRateLimitObserver {
+    /**
+     * Invoked whenever the rate limit changed. You should use {@link
+     * RateLimiter#isActiveForCategory(DataCategory)} to check whether the category you're
+     * interested in has changed.
+     *
+     * @param applied true if the rate limit was applied, false if it was lifted
+     */
+    void onRateLimitChanged(boolean applied);
```

Comment:
you suggest to use `RateLimiter.isActiveForCategory(DataCategory)` to check the limit
What's the reason of `applied` param? you don't seem to use it. 
Instead, we can pass the `RateLimiter` directly as parameter, so that you don't have to call the hub and handle nullable values on the call site. Wdyt?